### PR TITLE
Extract supervisor service from core

### PR DESCRIPTION
**Pull request by popzxc**
...and return where it belongs to.

Quick summary:

- `exonum-supervisor` is now a separate crate in the `services` directory.
- `blockchain` does not insert `exonum-supervisor` into services list.
- ...and `exonum-cli` does.
- `testkit` is adapted according to the changes above.
- `supervisor` tests from `testkit` are moved to the `exonum-supervisor` crate.

### CODE REVIEW

File: services/supervisor/src/errors.rs [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 /// Common errors emitted by transactions during execution.
 #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, IntoExecutionError)]
-#[exonum(crate = "crate")]
+#[exonum()]
```
[the remark this answers is not in the document]
Comment (popzxc) [reply services/supervisor/src/errors.rs]: I've decided to think about this macro like about some kind of magic that does *something* `¯\_(ツ)_/¯`

Here it ineed seems to be unneccessary.

File: services/supervisor/src/transactions.rs [hunk 2]
```diff
@@ -25,7 +25,7 @@ use crate::{
 use super::{DeployConfirmation, DeployRequest, Error, Schema, StartService, Supervisor};
 
 /// Supervisor service transactions.
-#[exonum_service(crate = "crate")]
+#[exonum_service()]
 pub trait Transactions {
```
Comment (vitvakatu) [inline services/supervisor/src/transactions.rs]: nit: Can we name it `SupervisorInterface`, as `cryptocurrency-advanced` example does?

File: testkit/src/lib.rs [hunk 3]
```diff
@@ -242,6 +243,19 @@ impl TestKit {
             .create()
     }
 
+    /// Returns the list of the Instances enabled by default.
+    ///
+    /// Currently only `Supervisor` service is included.
+    fn default_service_instances() -> Vec<InstanceCollection> {
```
Comment (alekseysidorov) [inline testkit/src/lib.rs]: In my PR with Configure interface I refuse of default service instances in Testkit. I think that you should completely remove the supervisor dependency in the testkit.

Comment (popzxc) [reply testkit/src/lib.rs]: Currently it's not really possible.
Let's wait until your PR will be merged then.

File: cli/src/lib.rs [hunk 4]
```diff
@@ -115,7 +116,7 @@ pub struct NodeBuilder {
 impl NodeBuilder {
     /// Creates new builder.
     pub fn new() -> Self {
-        Self::default()
+        Self::default().with_service(Supervisor)
```
Comment (alekseysidorov) [inline cli/src/lib.rs]: Are you sure that Supervisor will start with the default instance?

File: services/supervisor/src/lib.rs [hunk 5]
```diff
@@ -29,19 +34,20 @@ use crate::{
...
 mod transactions;
 
 #[derive(Debug, ServiceFactory)]
 #[exonum(
-    crate = "crate",
-    proto_sources = "proto::schema",
+    proto_sources = "proto",
     artifact_name = "exonum-supervisor",
-    implements("transactions::Transactions")
+    implements("transactions::SupervisorInterface")
```
Comment (alekseysidorov) [inline services/supervisor/src/lib.rs]: Why not always use `Transactions` interface name for the service own transactions?

Comment (popzxc) [reply services/supervisor/src/lib.rs]: [Try to agree between yourselves](https://github.com/exonum/exonum/pull/1473#discussion_r327947649)

Personally I don't care much about how it's called.

File: cli/src/lib.rs (hunk 4 above)
Comment (popzxc) [reply cli/src/lib.rs]: Good catch.

File: services/supervisor/src/lib.rs (hunk 5 above)
Comment (vitvakatu) [reply services/supervisor/src/lib.rs]: It's `<service-name>Interface` in every example. So let's be consistent in either way

**Pull request merged**